dynamiqs/mesolve/mesolve.py

```diff
@@ -1,25 +1,26 @@
 from __future__ import annotations
 
 from functools import partial
-from typing import Any, get_args
+from typing import Any
 
 import jax
 from jax import numpy as jnp
 from jaxtyping import ArrayLike
 
+from ..core._utils import _astimearray, get_solver_class
 from ..gradient import Gradient
 from ..options import Options
 from ..result import Result
 from ..solver import Dopri5, Euler, Solver
-from ..time_array import TimeArray, _factory_constant
+from ..time_array import TimeArray
 from ..utils.utils import todm
 from .mediffrax import MEDopri5, MEEuler
 
 
 @partial(jax.jit, static_argnames=('solver', 'gradient', 'options'))
 def mesolve(
     H: ArrayLike | TimeArray,
-    jump_ops: ArrayLike,
+    jump_ops: ArrayLike | list[ArrayLike | TimeArray],
```

Comment:
```suggestion
    jump_ops: ArrayLike | TimeArray | list[ArrayLike | TimeArray],
```

Reply:
I think we want `jump_ops` to be an iterable. If the user has a single time-dependent `jump_ops` he should pass a list with one element. The first `ArrayLike` typing is to account for a list of `jump_ops` already stacked. Maybe for clarity we can replace the `jump_ops` typing by the list only, and the `exp_ops` typing by a list too (even though it's a little imprecise, because it does not reflect that a stacked array can be passed, but it's much clearer that the expected object should have dimension `(nL, n, n)`).

Reply:
Agreed as well !